web/src/utils/search.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import { fetchV2 } from "./fetch";
+import { SearchResponse } from "@dzcode.io/api/dist/search/types";
+
+export const useSearch = (query: string, limit: number = 5) => {
+  const [results, setResults] = useState<SearchResponse["searchResults"]["results"]>();
```

Comment:
let it have an initial value:

```suggestion
  const [results, setResults] = useState<SearchResponse["searchResults"]["results"]>([]);
```

then we can get rid of fallbacks `(results || [])` in `web/src/components/search.tsx`